<!doctype html><html><head><meta charset=utf-8><meta http-equiv=x-ua-compatible content="ie=edge"><meta name=viewport content="width=device-width,initial-scale=1,shrink-to-fit=no"><title>Computational Systems Biology group &#183; CoLoMoTo</title><meta name=description content><link rel=stylesheet href=https://colomoto.github.io/style.main.css><link rel=stylesheet href=https://unpkg.com/leaflet@1.6.0/dist/leaflet.css integrity="sha512-xwE/Az9zrjBIphAcBb3F6JVqxf46+CDLwfLMHloNu6KEQCAWi6HcDUbeOfBIptF7tcCzusKFjFw2yuvEpDL9wQ==" crossorigin><script src=https://unpkg.com/leaflet@1.6.0/dist/leaflet.js integrity="sha512-gZwIG9x3wUXg2hdXF6+rVkLF/0Vi9U8D2Ntg4Ga5I5BZpVkVxlJWbSQtXPSiUTtC0TjtGOmxa1AJPuV0CPthew==" crossorigin></script></head><body><header><div class=title><a href=https://colomoto.github.io><img src=https://colomoto.github.io/colomoto_banner_white.svg class=logo alt=CoLoMoTo></a></div><nav><ul><li><a href=/>Home</a></li><li><a href=/members/>Members</a></li><li><a href=/news/>News / Events</a></li></ul></nav></header><section><div class=content><h1>Computational Systems Biology group</h1><p>Our team develop efficient tools to process functional genomic data and build computational models
of regulatory networks controlling mammalian cell differentiation.
We rely on the articulation of two approaches that are usually considered separately:
(i) a bioinformatic workflow to process epigenomic data and predict combinatorial transcriptional factor binding in promotors and enhancers;
(ii) a flexible logical modelling framework enabling the development of predictive models based on qualitative information,
e.g. regarding transcriptional factor binding and gene expression, potentially in the presence of genetic perturbations.
These two approaches are combined to decipher and model networks controlling mammalian cell fate, particularly focusing on blood cell specification, differentiation and reprogramming.</p><p>The first approach leans on the popular software RSAT (<a href=http://www.rsat.eu>http://www.rsat.eu</a>), which includes various algorithms for motif discovery and pattern matching, along with many utilities. We are currently extending this suite to support comparative and differential analyses of epigenomic data sets, including genome-wide DNA methylation and histone modification essays, in order to predict active enhancers, binding transcription factors and sites.</p><p>The second approach leans on the software GINsim (<a href=http://www.ginsim.org>http://www.ginsim.org</a>), dedicated to the definition and analysis of (multilevel) logical models of regulatory networks. To enable the analysis of larger networks (encompassing several hundreds of components), we are currently integrating advanced methods for model reduction, state transition graph compression, and model checking. We will further implement refinements of the logical framework through the association of probabilities with state transitions, to enable more quantitative simulations.</p><p>We are currently building a comprehensive model for the differentiation of T-helper cell into different canonical and hybrid subtypes in response to combinatorial signals (antigen presentation, co-stimulatory molecules, cytokines), in collaboration with the group of Dr Soumelis (Paris), who will provide systematic measurements (transcriptomics and proteomics).
In parallel, we are modelling the network controlling the specification of blood cells, primarily focusing on the myeloid and lymphoid pathways. In this respect, we rely on collaborations with Thomas Graf (Barcelona), Daniel Tenen (Singapore), and Ellen Rothenberg (Pasadena).
For both studies, we particularly focus on the articulation of three regulatory layers that are usually modelled separately: signalling pathways, transcriptional regulatory networks, and epigenetic remodelling mechanisms. We systematically exploit high-throughput data (from our partners or public repositories) to decipher the regulatory mechanisms (transcription factor binding at promoters and enhancers, epigenetic status) underlying functional effects (gene expression). We will progressively consider other functional data, starting with the roles of non coding RNAs known to affect blood cell fate.
Systematic confrontation of computational predictions against experimental data will foster model refinements until satisfactory recapitulation of cell responses to external signals and genetic perturbations. This then set the ground for predictive simulations of cell behaviour in novel situations, including responses to known drugs, with a emphasis on the delineation of efficient combinations of interventions, based on the prediction of synergistic effects of known drugs.</p></div></section><section><div class=content><b>Website:</b> <a href=http://www.ibens.ens.fr/spip.php?article26>http://www.ibens.ens.fr/spip.php?article26</a><br><b>Members involved in CoLoMoTo:</b><ul><li>Denis Thieffry</li><li>Aurélien Naldi</li><li>Céline Hernandez</li></ul><div id=map style=height:300px;width:300px></div><script>var map=L.map('map').setView(["48.842177"," 2.343683"],4);L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png',{attribution:'&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'}).addTo(map);L.marker(["48.842177"," 2.343683"]).addTo(map);</script></div></section></body></html>