<!doctype html><html><head><meta charset=utf-8><meta http-equiv=x-ua-compatible content="ie=edge"><meta name=viewport content="width=device-width,initial-scale=1,shrink-to-fit=no"><title>Workshop at [BC]2 2017 &#183; CoLoMoTo</title><meta name=description content><link rel=stylesheet href=https://colomoto.github.io/style.main.css></head><body><header><div class=title><a href=https://colomoto.github.io><img src=https://colomoto.github.io/colomoto_banner_white.svg class=logo alt=CoLoMoTo></a></div><nav><ul><li><a href=/>Home</a></li><li><a href=/members/>Members</a></li><li><a href=/events/>Events</a></li><li><a href=/news/>News</a></li></ul></nav></header><section><div class=content><h1>Workshop at [BC]2 2017</h1><p>The <a href=https://www.bc2.ch/2017/>13th [BC]2 - the Basel Computational Biology Conference</a> will take place on 12th-15th September 2017, in Basel, Switzerland.</p><p>[BC]2 will feature a one-day workshop/tutorial sessions, on <strong>Tuesday 12th of September</strong>, as part of the official [conference program])https://www.bc2.ch/2017/program).
These sessions will take place at the <strong>Kollegienhaus of the University of Basel</strong> (please consult the <a href=https://www.bc2.ch/2017/travel-venue>Location and Venue</a>).</p><p>Website: <a href=https://www.bc2.ch/2017/program/workshops/ws4>WS4: Logical modelling of biological regulatory networks</a></p><h2 id=organizers>Organizers</h2><p><a href=https://science.curie.fr/members/laurence-calzone/>Laurence Calzone</a> (Institut Curie / U900 INSERM / Mines ParisTech),
<a href=http://pedromonteiro.org>Pedro T. Monteiro</a> (INESC-ID/IST - Universidade de Lisboa),
<a href=https://www.vital-it.ch/about/team>Julien Dorier</a> (Vital-IT, SIB Swiss Institute of Bioinformatics).</p><h2 id=background>Background</h2><p>Logical models have long been used to explore the dynamical behaviours of regulatory networks. Over the past years, the types of available data that nourish the models or that the models need to explain has led to the development and the evolvement of modelling methods and tools in order to answer more specific problems. Indeed, logical modelling has become more and more reliant on transcriptomic, proteomic and epigenetic data.</p><p>The scope of this workshop is to provide an overview of the recent developments on methods and applications on the use of the logical formalism for the modelling of regulatory networks.</p><p>Following a community effort with the <a href=http://colomoto.org>Consortium for Logical Models and Tools</a>) to promote the exchange of models and to improve existing methods, we propose a workshop that will also expose the advances in the understanding of signalling pathways, cell differentiation, cell reprogramming or cell metabolism, etc.</p><p>We will cover the following three main topics:</p><ul><li>modelling tools: GINsim, CellNOpt, optimusqual, MaBoSS, CellCollective, EpiLog, etc;</li><li>current state of the art methods for static and dynamical analysis;</li><li>biological applications using logical modelling.</li></ul><h2 id=organization>Organization</h2><p>The workshop will be divided into two sessions: one session dedicated to recent developments of methods and tools; and another session dedicated to modelling applications to cell biology. Each session will be composed by one keynote presentation, followed by 5 presentations. At the end of the each session, we will devote some time for discussions with the audience.</p><h2 id=target-audience>Target audience</h2><p>This workshop should appeal to all students and researchers interested in learning about the recent developments on modelling regulatory networks, ranging from recent biological applications, to the development of methods and tools.</p><h2 id=program>Program</h2><h3 id=session-i---methodologies-and-tools>Session I - Methodologies and Tools</h3><table><thead><tr><th>Time</th><th>Speaker</th><th>Topic</th></tr></thead><tbody><tr><td>09:00</td><td><strong>Denis Thieffry</strong></td><td><strong>Predictive dynamical modelling of immune cell specification and reprogramming</strong></td></tr><tr><td>09:50</td><td>Loic Paulevé</td><td>Scalable and formal control of traces in qualitative networks with PINT</td></tr><tr><td>10:20</td><td></td><td>Coffee Break</td></tr><tr><td>10:50</td><td>Julien Dorier</td><td>Boolean regulatory networks reconstruction using literature based knowledge</td></tr><tr><td>11:20</td><td>Aurélien Naldi</td><td>Efficient identification of qualitative attractors through constraint solving</td></tr><tr><td>11:50</td><td>Pedro Monteiro</td><td>On the number and structure of Boolean functions compatible with a regulatory network topology</td></tr><tr><td>12:20</td><td></td><td>Discussion - concluding remarks</td></tr></tbody></table><h3 id=session-ii---applications-of-logical-models-to-diseases>Session II - Applications of logical models to diseases</h3><table><thead><tr><th>Time</th><th>Speaker</th><th>Topic</th></tr></thead><tbody><tr><td>13:30</td><td><strong>Ioannis Xenarios</strong></td><td><strong>Systems-level modeling between biocuration, predictions and validations</strong></td></tr><tr><td>14:15</td><td>Thomas Sauter</td><td>FALCON: A Toolbox for the Fast Contextualization of Logical Networks</td></tr><tr><td>14:45</td><td>Attila Gabor</td><td>Logic models of signalling networks & their training to experimental data with CellNOpt</td></tr><tr><td>15:05</td><td></td><td>Coffee Break</td></tr><tr><td>15:30</td><td>Gautier Stoll</td><td>Heterogeneous cell population modelling within continuous time Boolean approach</td></tr><tr><td>16:00</td><td>Claudine Chaouiya</td><td>A logical model of cancer cell adhesion properties in the context of Epithelial to Mesenchymal Transition</td></tr><tr><td>16:30</td><td></td><td>Discussion - concluding remarks</td></tr></tbody></table></div></section></body></html>