<!doctype html><html><head><meta charset=utf-8><meta http-equiv=x-ua-compatible content="ie=edge"><meta name=viewport content="width=device-width,initial-scale=1,shrink-to-fit=no"><title>Fourth CoLoMoTo meeting (Paris, July 17-19 2017) &#183; CoLoMoTo</title><meta name=description content><link rel=stylesheet href=https://colomoto.github.io/style.main.css></head><body><header><div class=title><a href=https://colomoto.github.io><img src=https://colomoto.github.io/colomoto_banner_white.svg class=logo alt=CoLoMoTo></a></div><nav><ul><li><a href=/>Home</a></li><li><a href=/members/>Members</a></li><li><a href=/events/>Events</a></li><li><a href=/news/>News</a></li></ul></nav></header><section><div class=content><h1>Fourth CoLoMoTo meeting (Paris, July 17-19 2017)</h1><p>The fourth CoLoMoTo meeting took place in Paris, from the 17th to the 19th of July, 2017.
As usual, the meeting covered a wide range of topics related to logical modelling,
from methodological and software developments to biological applications.
It was hosted by IBENS in the center of Paris (46 rue d&rsquo;Ulm).</p><p>This meeting was organized by
Aurélien Naldi (IBENS, Paris),
Claudine Chaouiya (IGC, Lisbon),
Denis Thieffry (IBENS, Paris),
Gautier Stoll (INSERM, IGR, U. Paris Descartes),
and Laurence Calzone (Curie Institute, Paris).</p><h2 id=july-17---scientific-presentations>July 17 - Scientific Presentations</h2><table><thead><tr><th>Time</th><th>Speaker</th><th>Topic</th></tr></thead><tbody><tr><td>13:00</td><td></td><td><strong>Introductory remarks</strong></td></tr><tr><td>13:20</td><td>Gautier Stoll</td><td><a href=Stoll_maboss.pdf>Using MaBoSS for modeling heterogeneous cell behavior</a></td></tr><tr><td>13:40</td><td>Samuel Collombet</td><td>Deciphering the regulatory network of blood cells specification and reprogramming</td></tr><tr><td>14:00</td><td></td><td><em>Coffee Break</em></td></tr><tr><td>14:30</td><td>Loïc Paulevé</td><td><a href=Pauleve_Pint.pdf>Modelling and analysis of transient dynamics of qualitative models</a> (<a href=pint-tutorial.pdf>tutorial</a>)</td></tr><tr><td>14:50</td><td>Enio Gjerga</td><td><a href=Gjerga_ILP.pdf>Efficient modelling of signalling networks derived from mass spectrometry phosphoproteomic data using Integer Linear Programming</a></td></tr><tr><td>15:10</td><td>Ping Zhang</td><td>Control and Diagnosis of Boolean Networks</td></tr><tr><td>15:30</td><td></td><td><em>Coffee Break</em></td></tr><tr><td>16:00</td><td>Anna Niarakis</td><td><a href=Niarakis_RA.pdf>Integrative modelling and analysis of molecular pathways involved in Rheumatoid Arthritis</a></td></tr><tr><td>16:20</td><td>Arnau Montagud</td><td><a href=Montagud_pipeline.pdf>Conceptual and computational framework for logical modelling of deregulated biological networks</a></td></tr><tr><td>16:40</td><td>Claudine Chaouiya</td><td><a href=Chaouiya_standards.pdf>Standards for logical models: current status of SBML qual & SED-ML</a></td></tr><tr><td>17:00</td><td></td><td><strong>Discussion - planning workgroups</strong></td></tr></tbody></table><h2 id=july-18---working-groups-and-presentations>July 18 - Working groups and Presentations</h2><table><thead><tr><th>Time</th><th>Speaker</th><th>Topic</th></tr></thead><tbody><tr><td>09:00</td><td></td><td><strong>Working group on tooling</strong></td></tr><tr><td></td><td>Aurélien Naldi</td><td><a href=Naldi_bioLQM.pdf>Model conversion, modification, and analysis with bioLQM</a></td></tr><tr><td></td><td>Loïc Paulevé</td><td>Jupyter Notebook</td></tr><tr><td></td><td>Claudine Chaouiya</td><td>Future extensions to SBML qual and SED ML</td></tr><tr><td>12:30</td><td></td><td><em>Lunch break</em></td></tr><tr><td>13:30</td><td></td><td><strong>Working group on applications</strong></td></tr><tr><td></td><td>Laurence Calzone</td><td>Model composition</td></tr><tr><td></td><td>Gautier Stoll</td><td>Modelling the immune system</td></tr><tr><td>15:30</td><td></td><td><strong>Working group on methods</strong></td></tr><tr><td></td><td>Pedro T. Monteiro</td><td><a href=Monteiro_functions.pdf>On the structure and robustness of gene regulatory network Boolean functions</a></td></tr><tr><td></td><td>Adrien Fauré</td><td><a href=Faure_circuits.pdf>A circuit-preserving mapping from multilevel to Boolean dynamics</a></td></tr><tr><td></td><td>Aurélien Naldi</td><td><a href=Naldi_trapspaces.pdf>Trap-spaces and global reachability mapping</a></td></tr><tr><td>17:30</td><td></td><td><strong>Wrapping-up</strong></td></tr></tbody></table><h2 id=july-19---wrapping-up-and-hacking>July 19 - Wrapping up and hacking</h2><table><thead><tr><th>Time</th><th>Topic</th></tr></thead><tbody><tr><td>10:00</td><td>MaBoSS Tutorial</td></tr><tr><td>10:00</td><td>Docker session</td></tr><tr><td>12:00</td><td><strong>Discussion and Wrapping-up</strong></td></tr></tbody></table><h1 id=summary-of-discussions>Summary of discussions</h1><h2 id=tool-integration--distribution>Tool integration & distribution</h2><p>Discussion moderated by Loïc Paulevé</p><p>Aim: share and reproduce computational systems biology workflows</p><p>Objectives:</p><ul><li>Provide a joint distribution of tools for computational systems biology, developed by CoLoMoTo members and others, using Docker image(s).</li><li>Provide (and advertise) an interface to write and document model analysis workflows, which typically rely on several tools, using the Jupyter notebook web interface.</li></ul><p>We also discussed teaching aspects, and potential publications.</p><h3 id=colomoto-docker>CoLoMoTo Docker</h3><p>Docker is used in many cloud environments, including data science and bioinformatics, to promote accessible and reproducible software (see for instance the BioContainers project).
Docker provides a container system, which makes easy to provide complete environments for running executables. On Linux, it does not require any virtualisation, hence have no performance drop. On Mac OSX and Windows, it can exploit latest virtualisation technologies embedded in processors to minimise the overhead.
Once the docker system is installed on the machine, installing and running a docker image can be done with a single command.</p><p>Our objective is to provide a docker image including the main software tools for manipulating and analysing logical models of biological networks. It may include, for instance, BioLQM, EpiLog, GINsim, clingo, MaBoSS, NuSMV, Pint, Boolsim, Node.JS, bns, Boolnet, espresso, etc.</p><p>Such an image will be useful in two main use cases:</p><ul><li>People can use this image to execute the provided tools avoiding installation/maintenance burden;</li><li>The docker image will also provide an interactive web interface to write and document model analysis workflows, for example using the Jupyter notebook.</li></ul><h3 id=actions-taken-status-of-the-project>Actions taken, status of the project</h3><ul><li>A github project for building the Docker image has been created by Loïc, Aurélien, Céline and Pedro at
<a href=https://github.com/colomoto/colomoto-docker>https://github.com/colomoto/colomoto-docker</a>.
The image currently includes GINsim, BioLQM, Pint, MaBoSS, NuSMV and NuSMV-ARCTL. More to come soon. The image also ships the Jupyter notebook.
Demos and examples will be produced in the coming months.</li><li>The docker image colomoto/colomoto-docker is available at <a href=https://hub.docker.com/r/colomoto/colomoto-docker>https://hub.docker.com/r/colomoto/colomoto-docker</a>.
It is automatically built from the github repository.</li><li>A documentation on how to include and maintain new tools will be added soon.</li><li>Future directions include the integration of systematic tests to ensure image consistency (test workflows / scripts).</li><li>Céline wrote a SnakeMake workflow using the docker image and a preliminary sketch of Jupyter notebook, accessible at
<a href=https://github.com/colomoto/colomoto-docker/tree/master/usecases>https://github.com/colomoto/colomoto-docker/tree/master/usecases</a>.</li></ul><h3 id=ipython-jupyter-notebook>IPython Jupyter Notebook</h3><p>Jupyter is an interactive web interface allowing to write notebooks combining code (python or R functions) and structured text for documentations.
See <a href=http://nbviewer.jupyter.org/github/pauleve/pint/blob/master/notebook/quick-tutorial.ipynb>http://nbviewer.jupyter.org/github/pauleve/pint/blob/master/notebook/quick-tutorial.ipynb</a>
for an example of notebook using GINsim and Pint.</p><p>A notebook consists in a single file that can shared and re-executed. Ultimately, one should be able to write a complete model workflow analysis in a single notebook, showing explicitly the full pipeline of tools and analysis parameters.</p><p>Although Python is the standard environment for Jupyter, a notebook can combine commands of different languages, including R.</p><p>Collectively, we aim to provide Python API for the tools shipped in the CoLoMoTo docker.
Writing a Python API for tools requires limited effort, as it may simply call the commands in the background with the appropriate parameters. The main work consists typically in returning python objects for the results, which enables further treatments of the results with other tools (through the python API), or with a standard library, for instance for data analysis and visualisation (e.g. using scipy, pandas, or matplotlib, just to name a few).</p><p>A typical workflow will often include a “manual” modification of a model, for instance, edge/node addition/removal. We could design an API to ease such modifications, e.g. through a dedicated python module to manipulate Boolean/multi-valued networks, or through some extension of the BioLQM script interface.</p><p>Other model manipulation to be considered include model merging, the handling of compartments, the integration of multicellular EpiLog grammar. These points would deserve further discussion.</p><p>User interface considerations: Jupyter allows to add custom menus and widgets (using javascript). Custom menus could help to launch pre-defined commands (see the Pint tutorial for an example), while custom widgets can ease the handling of command parameters.</p><p><em>Proposed action</em>: organize a 2-3 days hackathon with developers of CoLoMoTo tools to define and implement python interfaces, ideally still in 2017. Loïc will send a message to the mailing to start preparing it.</p><h3 id=demonstration>Demonstration</h3><p>The docker image and Jupyter notebook capabilities should be demonstrated on at least two different workflows combining different tools.
Workflow proposals:</p><ul><li>Model + mutations -> qualitative simulations -> stochastic simulations: may rely on BioLQM, GINsim, and MaBoSS.
Candidate models: Macrophages ; T-Cell differentiation ; TNF</li><li>Rough model -> verification and filtering with stable states and model-checking: may rely on BioLQM, GINsim, Pint.
Candidate models: Cell cycle ; EMT ; MAPK</li></ul><p>Inspiration: SBML-qual paper, Curie workflow (Arnau Montagud), PNAS paper (Samuel Collombet).</p><h3 id=teaching-aspect>Teaching aspect</h3><p>The question of the usage of Docker and Jupyter notebook in a teaching class environment has been raised by Denis.
Requiring students to install docker and pull the image (which may weigh several hundreds of megabytes) does not seem tractable.
However, one could rely on cloud infrastructures supporting Docker to provide an easy access to the tools: the student would simply have to connect to a given URL in order to write workflows through the Jupyter web interface.
In France, IFB can deploy Docker images and provide resources and access to students (this has been done already for courses using Jupyter notebook for bioinformatic tutorials).</p><h3 id=publication>Publication</h3><p>A paper for promoting the docker and the Jupyter notebook to ease the writing and sharing of computational systems biology workflows could be sent to Frontiers Research Topic on Logical Modeling of Cellular Processes: From Software Development to Network Dynamics (abstract 1st September, submission 1st February).
The paper may show an instantiation of the docker and Jupyter notebook on two workflows.</p><p>“Screenshots” of the discussion:</p><p><img src=distribution_1.jpeg alt>
<img src=distribution_2.jpeg alt>
<img src=distribution_3.jpeg alt></p><h2 id=sbml-qual-and-sed-ml>SBML-qual and SED-ML</h2><p>Discussion moderated by Claudine Chaouiya</p><h3 id=sbml-qual>SBML-qual</h3><ul><li><p>Test suite: We need to contribute with a test suite for SBML-qual (see <a href=http://sbml.org/Software/SBML_Test_Suite)>http://sbml.org/Software/SBML_Test_Suite)</a>.
This would fulfil the need for a service validating the correctness of interpretation of SBML-qual files.</p></li><li><p>Support of Layout package: F. Bergmann (contact for the SBML-layout package) will be contacted to ask for the support of SBML-qual
(i.e. QualitativeSpecies as a minimal requirement, we need to check what is needed for the interactions).</p></li><li><p>Extensions of SBML-qual format</p><ul><li>support for incomplete models as such,</li><li>support for families of models,</li><li>support for probabilities associated with the transitions (on top of the previous point?), to cover Probabilistic models à la Shmulevich.</li></ul></li></ul><p>A number of other possible extensions were discussed, but it was agreed to postpone their consideration.</p><h3 id=sed-ml>SED-ML</h3><p>It was agreed that simulation is one type of model analysis. For example, we will need to cover:</p><ul><li><p>Identification of stable states or trap spaces (independent of the updating scheme).</p></li><li><p>Identification of attractors (dependent of the updating scheme, as cyclic attractors).</p></li><li><p>Reachability properties requiring the specification of :</p><ul><li>(set of) initial state(s) but also (set) of target state(s),</li><li>temporal properties (LTL and CTL).</li></ul></li><li><p>Probabilities associated with initial states.</p></li><li><p>Definition of the behaviour of input components: e.g. (range of) probabilities associated with their values, switching values under specific conditions (see point 3.b), etc.</p></li><li><p>Definition of model perturbations:</p><ul><li>Override the ´model modification´as defined in SED-ML or not? The point is that it would be more convenient to rely on
a simpler description of model perturbation: block the value of a node within an interval, block the value of a regulatory within an interval.</li><li>Apply model perturbations at a given simulation step (this is applicable for specific simulation settings such as synchronous or random asynchronous).</li><li>Make model perturbations stochastic, by associating a probability.</li></ul></li><li><p>In general, would it be possible to switch simulation parameters when reaching a state compliant with a certain condition (see point 3.b), or at a given step?</p></li><li><p>For (part of) these settings, we may need to add new terms to the KISAO ontology, check if there is the notion of exact and estimated results,
as some setting should lead always to the same specific results, other not (stochastic simulations).</p></li></ul><p><img src=sedml.jpeg alt></p><h2 id=model-composition>Model composition</h2><p>Discussion moderated by Laurence Calzone</p><p>Individual models have already been developed and validated. They can be found in the repository of GINsim, in CellCollective, and to a certain extent in BioModels. These individual models can be seen as modules of “cell cycle”, “apoptosis”, “DNA repair”, etc. The question is: when creating a new model, could we use existing models and, if so, how to assemble them?</p><p>There are two types of possible compositions:</p><ul><li>intracellular: pathways (or modules) within the cell,</li><li>intercellular: different cell types interacting, or a population of interacting cells of the same type.</li></ul><p>The idea of composing a model from already existing logical models poses several questions:</p><ul><li>how to deal with name conflicts: two models may have different naming for the same entity;</li><li>how to deal with overlap: which part to use when two models have two different ways of writing rules;</li><li>a model may provide details hidden behind a single interaction in the other model:
EGFR in EMT model has a role in invasion and in cell cycle proliferation, which depends on the context;</li><li>how to integrate external signals;</li><li>how to preserve layout, which constitutes an important piece of information when constructing a regulatory network.</li></ul><p>We are aware that a software tool to compose models cannot fully automate the merging, but it should ease merging and suggest options for pending issues.</p><p>We could get some inspiration from the SBML-comp package to define the rules for composing two models.</p><p>To ensure the construction of comprehensive regulatory graph, when merging two models, we could test using OmniPath that all the connections between the nodes, that are found in other databases, are added to the composed model and annotated with publications.</p><p>The following proposed workflow/objectives emerged from the discussion:</p><ul><li>identify common components (referred to as interface nodes);</li><li>keep track of the original rules;</li><li>generate a default rule (i.e. join the original functions with an OR function?) and let the user edit the new rules if necessary;</li><li>remember the modeller’s decision to formally define a merging strategy: keep track of all decisions to be able to come back to these
rules if something does not fit in the model, or allow to change the rules for which the modeller was not confident at first
when checking the sensitivity of the model.</li></ul><p>Note that EpiLog has a grammar for simple integration functions, which could be extended to cope for more complex cases.</p><h2 id=modelling-the-immune-system>Modelling the immune system</h2><p>Discussion moderated by Gautier Stoll.</p><p>Immune system is a complex machinery, implying multiple signalling pathways in an heterogeneous environment. Understanding (part of) immune is a very important challenge in biology, with a lot of potential medical applications (infection, autoimmune diseases, cancer, etc.)</p><h3 id=immune-system-parts-that-are-modeled>Immune system parts that are modeled</h3><p>Modeling the full immune system (including cell types and genes/proteins) is probably out of reach with the available tools. Nevertheless, some members of CoLoMoTo consortium are currently modelling subparts of the immune system:</p><ul><li>Th cells differentiation (Th, Treg, etc.),</li><li>DC differentiation,</li><li>Th cells activation and immune checkpoints,</li><li>TCR (alpha+beta+) T cell selection.</li></ul><h3 id=phenotypes>Phenotypes</h3><p>It can be interesting to describe which phenotypes can be studied with the Logical Modelling approach:</p><ul><li>cell differentiation,</li><li>cell selection (through cell death and proliferation),</li><li>ligand ← → receptor interaction,</li><li>spatial distribution; to keep the logical modeling philosophy, spatial effects could be modeled within discrete compartments.</li></ul><p>One question arises: are these “phenotypes” really specific to the immune system? The answer is no, as each of them could be found in other biological processes. But it remains interesting to regroup immune system models, at least because modelers may have specific problems/results to share.</p><h3 id=experimental-data>Experimental data</h3><p>There are specific methods for obtaining data to characterize the immune system:</p><ul><li>single cell characterization (cytometry, single cell RNA seq, etc.)</li><li>transcriptome (micro-arrays, RNA-seq) within immune infiltrate estimation; this is an indirect method, for which different bioinformatics tools exist;</li><li>&mldr;</li></ul><h3 id=methods>Methods</h3><p>Logical Modelling widely applied to signaling pathways. For the immune system, modelers use the following approaches:</p><ul><li>Boolean modeling and attractor identification,</li><li>Probabilistic Boolean networks,</li><li>Agent-based modeling.</li></ul><h1 id=participants>Participants</h1><ul><li>Adrien Fauré (Yamaguchi University)</li><li>Anna Niarakis (Université d&rsquo;Evry Val d&rsquo;Essonne)</li><li>Arnau Montagud (Institut Curie)</li><li>Aurélien Naldi (IBENS)</li><li>Célia Biane (UEVE, IBISC Lab)</li><li>Céline Hernandez (IBENS)</li><li>Claudine Chaouiya (Instituto Gulbenkian de Ciência)</li><li>Daniela Ferretti (University of Camerino)</li><li>Delphine Potier (CIML)</li><li>Denis Thieffry (IBENS)</li><li>Élisabeth Remy (CNRS)</li><li>Enio Gjerga (JRC COMBINE, RWTH Aachen University)</li><li>Gautier Stoll (INSERM U1138eq11/Université Paris 5/Gustave Roussy)</li><li>Geoffray Brelurut (IBENS)</li><li>Ghislaine Morvan-Dubois (CNRS)</li><li>Hugues Mandon (INRIA - LSV - LRI)</li><li>Laurence Calzone (Institut Curie)</li><li>Loïc Paulevé (CNRS/LRI)</li><li>Nathalie Lehmann (IBENS)</li><li>Paolo Ballarini (CentraleSupelec)</li><li>Pauline Traynard (Institut Curie)</li><li>Pedro T. Monteiro (INESC-ID/IST - Universidade de Lisboa)</li><li>Ping Zhang (University of Kaiserslautern, Institute of Automatic Control)</li><li>Samuel Collombet (IBENS)</li><li>Samyadeep Basu (IBENS)</li><li>Seraya Maouche (TBi Scientific)</li><li>Swann Flochlay (IBENS)</li><li>Tomas Helikar (University of Nebraska)</li><li>Vidisha Singh (Genhotel, University of Evry val d&rsquo;essonne)</li><li>Wassim Abou-Jaoude</li></ul></div></section></body></html>